src/AdminConsole/Endpoints/ApplicationEndpoints.cs

```diff
@@ -13,6 +17,10 @@ public static IEndpointRouteBuilder MapApplicationEndpoints(this IEndpointRouteB
 
         // The Blazor SSR sample uses the same solution for signing out, but we do not want to use all the endpoints.
         group.MapGet("/{appId}/Available", IsAppIdAvailableAsync).RequireAuthorization();
+
+        group.MapGet("/{appId}/DownloadOnboarding", DownloadOnboardingAsync)
+            .RequireAuthorization(CustomPolicy.HasAppRole);
```

Comment:
The `HasAppRole` policy is the same one that checks if we're allowed to access an application.